app/src/main/java/com/github/epfl/meili/map/PoiRenderer.kt

```diff
@@ -0,0 +1,50 @@
+package com.github.epfl.meili.map
+
+import android.content.Context
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.model.BitmapDescriptor
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
+import com.google.android.gms.maps.model.MarkerOptions
+import com.google.maps.android.clustering.ClusterManager
+import com.google.maps.android.clustering.view.DefaultClusterRenderer
+
+/**
+ * PoiRenderer is the class which renders POIs as Markers on the Map and also handles cluster rendering
+ */
+open class PoiRenderer(context: Context?, map: GoogleMap?, val clusterManager: ClusterManager<PoiItem>)
+    : DefaultClusterRenderer<PoiItem>(context, map, clusterManager) {
+
+    private var poiStatusMap: Map<PoiItem, PoiMarkerViewModel.PointOfInterestStatus>? = null
+
+    override fun onBeforeClusterItemRendered(item: PoiItem, markerOptions: MarkerOptions) {
+        val icon: BitmapDescriptor
+        if (poiStatusMap == null || !poiStatusMap!!.contains(item)) {
+            icon = DEFAULT_ICON
+        } else {
+            when (poiStatusMap!![item]) {
```

Comment:
Nitpick: lift the icon assignment out of the when

Reply:
Awesome! Didn't know about this. Thanks!